### DISCUSSION TITLE
Flappdoodle mongo using random ports

### DISCUSSION THREAD

**Discussion by amumuney** [flapdoodle-oss/de.flapdoodle.embed.mongo#456] [Q&A] [answered]
When flapdoodle embedded mongo starts, it does so on random ports. I want to be able to start on a specified port. I have tried the code below to no avail. Please kindly advise in this regard. Thank you.

```
final Distribution distribution = Distribution.of(Version.Main.PRODUCTION, ImmutablePlatform.builder()
                .operatingSystem(OS.Linux).architecture(CommonArchitecture.X86_64).version(CentosVersion.CentOS_8).build());

        final ImmutableMongodArguments immutableMongodArguments = MongodArguments.defaults().withUseNoPrealloc(false)
                .withUseSmallFiles(false);
        ImmutableNet immutableNet = Net.builder().bindIp("127.0.0.1").port(27017).isIpv6(Network.localhostIsIPv6()).build();
        final Mongod exe = Mongod.instance()
//                .withMongodProcessArguments(MongodProcessArguments.builder().net(StateID.of("net",Net.class)).build())
                .withMongodArguments(Start.to(MongodArguments.class).initializedWith(immutableMongodArguments))
                .withDistribution(Start.to(Distribution.class).initializedWith(distribution))
                .withNet(Start.to(Net.class).initializedWith(immutableNet));
        TransitionWalker.ReachedState<RunningMongodProcess> running = exe.start(Version.Main.PRODUCTION);
        ```

**Comment by michaelmosmann** [comment]
@amumuney do you need still an answer to this?

**Reply by amumuney** [reply]
@michaelmosmann Yes Please.

**Reply by michaelmosmann** [reply, marked as answer]
@amumuney now you can find this in the documentation: https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo/blob/main/docs/Howto.md#-or-with-fixed-value .. so you are on the right track.. except for this:

`MongodProcessArguments.builder().net(StateID.of("net",Net.class)).build()`

.. just remove this and it should work. You can further simplify this by removing `distribution` and change this by setting an system property.. see https://github.com/flapdoodle-oss/de.flapdoodle.os#run

**Reply by amumuney** [reply]
Thanks. It works. @michaelmosmann
